File: src/test/java/com/tambapps/math/complex/ComplexTest.java
package com.tambapps.math.complex;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

import org.junit.Test;

public class ComplexTest {

  @Test
  public void addTest() {
    Complex complex1 = Complex.of(8);
    Complex complex2 = Complex.of(4, 1);

    assertEquals("Should be equal", Complex.of(12, 1), complex1.plus(complex2));
  }

  @Test
  public void pureTest() {
    Complex pr = Complex.of(3423);
    Complex pi = Complex.of(0, 234);

    assertTrue("Should be only pure real", pr.isPureReal() && !pr.isPureImaginary());
    assertTrue("Should be only pure imaginary", pi.isPureImaginary() && !pi.isPureReal());
  }

  @Test
  public void mulTest() {
    Complex c1 = Complex.ONE;
    Complex c2 = Complex.of(0, 2);

    assertEquals("Should be equal", Complex.ZERO, c1.multiply(Complex.ZERO));
    assertEquals("Should be equal", Complex.ZERO, Complex.ZERO.multiply(c1));

    assertEquals("Should be equal", Complex.I.multiply(2), c1.multiply(c2));
    assertEquals("Should be equal", Complex.I.multiply(2), c2.multiply(c1));
  }

}

File: src/test/kotlin/com/tambapps/fft4j/complex/ComplexTest.kt
package com.tambapps.fft4j.complex

import junit.framework.Assert.assertEquals
import org.junit.Assert
import org.junit.Test

class ComplexTest {

    @Test
    fun addTest() {
        val complex1 = Complex.of(8.0)
        val complex2 = Complex.of(4.0, 1.0)
        assertEquals("Should be equal", Complex.of(12.0, 1.0), complex1.plus(complex2))
    }

    @Test
    fun pureTest() {
        val pr = Complex.of(3423.0)
        val pi = Complex.of(0.0, 234.0)
        Assert.assertTrue("Should be only pure real", pr.isPureReal && !pr.isPureImaginary)
        Assert.assertTrue("Should be only pure imaginary", pi.isPureImaginary && !pi.isPureReal)
    }

    @Test
    fun mulTest() {
        val c1 = Complex.ONE
        val c2 = Complex.of(0.0, 2.0)
        assertEquals("Should be equal", Complex.ZERO, c1 * Complex.ZERO)
        assertEquals("Should be equal", Complex.ZERO, Complex.ZERO * c1)
        assertEquals("Should be equal", Complex.I * 2.0, c1 * c2)
        assertEquals("Should be equal", Complex.I * 2.0, c2 * c1)
    }

}
